SwiftBuddiesIOS/Targets/LoginModule/Sources/AuthenticationViewModel.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  AuthenticationViewModel.swift
+//  Login
+//
+//  Created by Berkay Tuncel on 19.04.2024.
+//
+
+import Foundation
+import Auth
+
+@MainActor
+final class AuthenticationViewModel: ObservableObject {
+    
+    func signInGoogle() async throws {
+        let helper = SignInGoogleHelper()
+        let tokens = try await helper.signIn()
+        let _ = try await AuthenticationManager.shared.signInWithGoogle(tokens: tokens)
```

Comment:
buradaki akista gerekli viewmodel sadece authentication manager'a erismeli ve o da inject edilmeli, buradaki token logic'inin viewmodel'in icerisinde olmasina gerek yok

Reply:
aynisi diger signIn metodlari icin de gecerli